File: languages/java/cpu-prime-count/PrimeCount.java
public class PrimeCount {
    public static int countPrimes(int limit) {
        if (limit < 2) return 0;
        boolean[] sieve = new boolean[limit + 1];
        for (int i = 2; i <= limit; i++) sieve[i] = true;
        for (int p = 2; p * p <= limit; p++) {
            if (sieve[p]) {
                for (int multiple = p * p; multiple <= limit; multiple += p) {
                    sieve[multiple] = false;
                }
            }
        }
        int count = 0;
        for (int i = 2; i <= limit; i++) {
            if (sieve[i]) count++;
        }
        return count;
    }

    public static void main(String[] args) {
        int upper = 100000;
        if (args.length > 0) {
            upper = Integer.parseInt(args[0]);
        }
        System.out.println(countPrimes(upper));
    }
}

File: languages/typescript/cpu-prime-count/prime_count.mjs
#!/usr/bin/env node
const upper = Number(process.argv[2] || 100000);

function countPrimes(limit) {
  if (limit < 2) return 0;
  const sieve = new Uint8Array(limit + 1);
  sieve.fill(1);
  sieve[0] = 0;
  sieve[1] = 0;
  for (let p = 2; p * p <= limit; p += 1) {
    if (sieve[p]) {
      for (let multiple = p * p; multiple <= limit; multiple += p) {
        sieve[multiple] = 0;
      }
    }
  }
  let count = 0;
  for (let i = 2; i <= limit; i += 1) {
    count += sieve[i];
  }
  return count;
}

console.log(countPrimes(upper));
